lib/auth/sessions.go

```diff
@@ -609,7 +609,17 @@ func (a *Server) CreateAppSessionFromReq(ctx context.Context, req NewAppSessionR
 	log.Debugf("Generated application web session for %v with TTL %v.", req.User, req.SessionTTL)
 	UserLoginCount.Inc()
 
-	userMetadata := req.Identity.GetUserMetadata()
+	// Extract the identity of the user from the certificate.
```

Comment:
The "what" is sort of obvious from the lines below, but it's the "why" that is important here. Can you please explain in the comment why we're doing it this way?

A test would also be great to make sure we don't fall into this regression anymore.